### DISCUSSION TITLE
Kitten hints

### DISCUSSION THREAD

**Discussion by dpetka2001** [kovidgoyal/kitty#7040] [Q&A] [answered]
I have a keymap in my configuration to paste the git commit hashes of `git log` output to my terminal.

The keymap is this `map ctrl+shift+f kitten hints --program - --type regex --regex "(?ms)^[*]?\s(\S+)"`. Until kitty 0.31 this used to work. With latest upgrade to kitty 0.32 it doesn't any more. I get a `Error: No matches found`.

Did something change with the latest update and I should maybe change the regex expression I use?

**Comment by kovidgoyal** [comment]
There are no changes to the hints kitten in this release as can be easily verified by
looking at the history of the kittens/hints folder.

And I get no matches found for that expression with kitty 0.31.0 as well

**Reply by dpetka2001** [reply]
Seems to be the same as #6228. I had the same problem then.

**Reply by dpetka2001** [reply]
![2024-01-22_13-35](https://github.com/kovidgoyal/kitty/assets/12776461/5eefc146-8471-4582-874a-a4ee1463690f)
It works for me in kitty 0.31 as the screenshot shows.

**Reply by kovidgoyal** [reply]
Ah you arent matching normal git log output, with that I can reproduce. Must be caused by a change to the go regex2 package which was updated, I will look into it when I have a moment.

**Reply by dpetka2001** [reply]
Yes the command I use for git log is `git log --color --graph --date=format:'%Y-%m-%d %H:%M:%S' --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%ad) %C(bold blue)<%an>%Creset'` as I was also using previously in the issue that I mentioned. I will wait for your input whenever you have time. For the time being I'll just keep using kitty 0.31. 

Thank you very much.

**Reply by kovidgoyal** [reply, marked as answer]
Ah no its caused by a change in how arguments to map are parsed. They now use shell like parsing. that means you need to use 
```
--regex '(?ms)^[*]?\s(\S+)'
```
Note the single quotes instead of double quotes. Or if you want to use doubel quotes you have to double the backslashes.

**Reply by dpetka2001** [reply]
Can confirm with this change it works as expected. 

Really appreciate your help. Thank you very much.
